app/controllers/api/mixins/genealogy.rb

```diff
@@ -0,0 +1,58 @@
+module Api
+  module Mixins
+    module Genealogy
+      RELATIONSHIP_COLLECTIONS = %w[vms templates].freeze
+      VALID_EDIT_ATTRS = %w[description name child_resources parent_resource].freeze
+
+      def tag_specified(id, data)
+        if id.to_i > 0
+          klass  = collection_class(:tags)
+          tagobj = klass.find(id)
+          return tag_path_to_spec(tagobj.name).merge(:id => tagobj.id)
+        end
+        parse_tag(data)
+      end
+
+      def assign_genealogy(type, id, data)
```

Comment:
Not sure I like this name because it doesn't only assign genealogy...it also updates the rest of the attributes.  What about:

```suggestion
      def edit_resource_with_genealogy(type, id, data)
```

Reply:
Alternatively we can just have `edit_resource` and call super, but I'm concerned that might get "lost" in the mixin and confuse people.  Curious what @kbrock thinks

Reply:
your name looks good. I also like the way Gilbert's new code reads.
Good suggestion